### PR TITLE
Temporal control documentation starting (intro and raster)

### PR DESCRIPTION
**Pull request by ghtmtt**
Initial draft before to really start documenting this feature. 

I thought the structure could be this (given also what we discussed in #5443)

I also thing that it could be easier to make small PR and merge them step by step rather than having a long text change with a lot of new information.

### CODE REVIEW

**Comment by DelazJ** [comment]
> @ghtmtt requested your review on this pull request
>[DRAFT]

@ghtmtt I understand you want to do it step by step but is the current state still a draft, an ongoing work or the end of the first step requiring feedback?

**Comment by ghtmtt** [comment]
> @ghtmtt I understand you want to do it step by step but is the current state still a draft, an ongoing work or the end of the first step requiring feedback?

yep exactly. I will push something that you can review and that we can merge very soon :)

**Comment by ghtmtt** [comment]
@DelazJ @rduivenvoorde finally I got something on the temporal control :) let me know what you think about it

**Comment by DelazJ** [comment]
hmm... A rebase @ghtmtt ?

**Comment by ghtmtt** [comment]
@DelazJ if it is too confusing I can open a new PR

**Comment by DelazJ** [comment]
```
git checkout yourbranch
git pull --rebase upstream master 
```

should do the trick, I think

**Comment by ghtmtt** [comment]
thanks :)

**Comment by ghtmtt** [comment]
@DelazJ I think this PR is ready to (eventually) be merged. Other parts (vectors, mesh, expressions, etc) can be added in a second moment when we agree on the structure of the chapter.

File: docs/user_manual/temporal/temporal.rst [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+Furthermore, many variables are added to the Field Calculator so that you can
+use them to change the appearance of the layer(s) depending on the current map
+frame.
+
+
+General concepts
+================
+Temporal control settings are available at different levels:
+
+* in the `Temporal` tab within the project properties :menuselection:`Project --> Properties`
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:37]:
```suggestion
* in the :guilabel:`Temporal` tab within the :menuselection:`Project --> Properties` dialog
```
to avoid repetition

File: docs/user_manual/temporal/temporal.rst [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+   :align: center
+
+   Raster Layer Temporal Settings
+
+In this example the raster layer has a ``Start date`` set to ``2013-06-30 00:00:00``
+and no ``End Date``. It means that the raster will be convered by other rasters
+that have a more recent start date set.
+
+Once you have enabled the temporal control on the layer the
+|indicatorTemporalFixedRange| icon will be dislayed next to the layer name in
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:82]:
```suggestion
|indicatorTemporalFixedRange| icon will be displayed next to the layer name in
```

File: docs/user_manual/temporal/temporal.rst [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+Once you have set the temporal options you can animate the map with the |play|
+button. Use all the other buttons to pause, stop, rewind and forward the map:
+
+.. figure:: img/play_bar.png
+   :align: center
+
+   Animated Temporal Navigation Buttons
+
+If the |unchecked| :guilabel:`Loop` checkbox in the upper right part of the bar
+if checked the animation will be endlessly repeated.
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:120]:
```suggestion
is checked the animation will be endlessly repeated.
```

File: docs/user_manual/temporal/temporal.rst [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+If you have many different raster layers loaded in the legend be sure to tweak
+the temporal settings for each layer. In the following example the name of the
+layers describes the date.
+
+If you open the main temporal panel with the |temporal| button you can animate
+the map with by clicking the |indicatorTemporalAnimated| :guilabel:`Animated
+tamporal navigation` button.
+
+Before to play the map you have to set the temporal options shown in the bottom
+part of the ``Animated temporal navigation`` panel:
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:101]:
```suggestion
part of the :guilabel:`Animated temporal navigation` panel:
```

File: docs/user_manual/temporal/temporal.rst [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+* ``Range``: the starting and ending datetime range
+* ``Step``: how often the map should update
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:104]:
```suggestion
* :guilabel:`Range`: the starting and ending datetime range
* :guilabel:`Step`: how often the map should update
```

File: docs/user_manual/temporal/temporal.rst [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+Once you have enabled the temporal control on the layer the
+|indicatorTemporalFixedRange| icon will be dislayed next to the layer name in
+the legend to remind you that the layer is temporal controlled:
+
+.. figure:: img/temporal_icon_legend.png
+   :align: center
+
+   Temporal icon in the legend
+
+You can click on the icon to quickly go to the temporal layer settings.
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:90]:
```suggestion
You can click on the icon to quickly go to the temporal settings of the layer.
```
instead?

File: docs/user_manual/temporal/temporal.rst [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+General concepts
+================
+Temporal control settings are available at different levels:
+
+* in the `Temporal` tab within the project properties :menuselection:`Project --> Properties`
+* in the properties on the single layer(s)
+
+The whole temporal control is activated with the |temporal| icon located in the
+main toolbar. Once clicked the temporal panel will appear:
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:41]:
```suggestion
:guilabel:`Map Navigation` toolbar. Once clicked the temporal panel will appear:
```

File: docs/user_manual/temporal/temporal.rst [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+****************
+Temporal Support
+****************
+
+.. only:: html
+
+   .. contents::
+      :local:
+
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:11]: Can we consider adding a short description of what temporal data or temporal control means? And how this information can be stored? We usually do this when we introduce new "concepts" (see mesh, vector, raster, vector tile (soon)).

File: docs/user_manual/temporal/temporal.rst [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+   :align: center
+
+   Temporal Controller Panel
+
+The temporal panel has 3 main icons:
+
+* |indicatorTemporalOff| :guilabel:`Turn off temporal navigation`: all the temporal 
+  settings are diabled
+* |indicatorTemporalFixedRange| :guilabel:`Fixed range tamporal navigation`: a 
+  fixed time range is set and oly layers that fit this range will be diplayed in
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:55]:
```suggestion
  settings are disabled
* |indicatorTemporalFixedRange| :guilabel:`Fixed range temporal navigation`: a 
  fixed time range is set and only layers that fit this range will be displayed in
```

File: docs/user_manual/temporal/temporal.rst [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+the legend to remind you that the layer is temporal controlled:
+
+.. figure:: img/temporal_icon_legend.png
+   :align: center
+
+   Temporal icon in the legend
+
+You can click on the icon to quickly go to the temporal layer settings.
+
+If you have many different raster layers loaded in the legend be sure to tweak
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:92]:
```suggestion
If you have many different raster layers loaded in the project be sure to tweak
```

File: docs/user_manual/temporal/temporal.rst [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+the temporal settings for each layer. In the following example the name of the
+layers describes the date.
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:94]: could we provide some more info on the date?
```suggestion
layers describes the date at which they are set to display.
```
or something like that...

File: substitutions.txt [hunk 12]
```diff
@@ -537,6 +537,12 @@
    :width: 1.5em
 .. |indicatorTemporal| image:: /static/common/mIndicatorTemporal.png
    :width: 1.5em
+.. |indicatorTemporalOff| image:: /static/common/mTemporalNavigationOff.png
+   :width: 1.5em
+.. |indicatorTemporalFixedRange| image:: /static/common/mTemporalNavigationFixedRange.png
+   :width: 1.5em
+.. |indicatorTemporalAnimated| image:: /static/common/mTemporalNavigationAnimated.png
+   :width: 1.5em
```
[the remark this answers is not in the document]
Comment (DelazJ) [reply substitutions.txt:545]: And could you display them in the guidelines substitutions page?

File: docs/user_manual/temporal/temporal.rst [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+use them to change the appearance of the layer(s) depending on the current map
+frame.
+
+
+General concepts
+================
+Temporal control settings are available at different levels:
+
+* in the `Temporal` tab within the project properties :menuselection:`Project --> Properties`
+* in the properties on the single layer(s)
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:38]: And what is, if any, the relation between the project level temporal settings and the layer level's?
I think this is where we would explain the QGIS "concepts" as mentioned in the section title.

File: docs/user_manual/temporal/temporal.rst [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+* mesh layers
+* WMST
+
+Activating the temporal control support means setting up the temporal options on
+the layer(s) and then enabling the temporal view QGIS-wide.
+
+The temporal support is also available in the print layout: you can export time
+based maps.
+
+Furthermore, many variables are added to the Field Calculator so that you can
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:28]: the field calculator is quite restrictive. Better refer to expressions....

File: docs/user_manual/temporal/temporal.rst [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+
+.. _raster_temporal:
+
+Raster Layers
+=============
+Normally raster layers don't have any temporal metadata set. Therefore you have
+to choose layer by layer the temporal settings like the start date and
+eventually the end date:
+
+.. figure:: img/raster_layer_temporal_settings_1.png
```
Comment (DelazJ) [inline docs/user_manual/temporal/temporal.rst:72]: Any chance you could remove the background (hide the layer before shooting the dialog)? It's quite ugly this way.

**Review by DelazJ — COMMENTED** [review]
Hi @ghtmtt I made a review but consider this as JUST a cosmetic one. I don't use temporal settings and expected to level up with the PR, but I'm quite lost and find the read hard to follow.
If someone else wants to give it a look. Maybe it's me.

**Comment by DelazJ** [comment]
Ah I forgot to say: I like the gif at the end. and I wonder if it should not be part of the introduction (or somewhere at the beginning, to show how we can interact with/handle temporal data within QGIS)

File: docs/user_manual/temporal/temporal.rst [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+
+QGIS can handle temporal control on loaded layers depending on the information
+of the data. 
+
+Different providers are supported:
+
+* vector layers
+* raster layers
+* mesh
+* WMST
```
[the remark this answers is not in the document]
Comment (ghtmtt) [reply docs/user_manual/temporal/temporal.rst:20]: tihs is actually wms-t (wmts are wms tile based wms-t are wms with time data)

File: substitutions.txt (hunk 12 above)
[the remark this answers is not in the document]
Comment (ghtmtt) [reply substitutions.txt]: done. I created a new `Temporal` section, might be overkilled right now, but maybe in the future it is better to have a separated section. No strong opinions on moving it within an existing section

File: docs/user_manual/temporal/temporal.rst [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+
+The temporal panel has 3 main icons:
+
+* |indicatorTemporalOff| :guilabel:`Turn off temporal navigation`: all the temporal 
+  settings are diabled
+* |indicatorTemporalFixedRange| :guilabel:`Fixed range tamporal navigation`: a 
+  fixed time range is set and oly layers that fit this range will be diplayed in 
+  the map
+* |indicatorTemporalAnimated| :guilabel:`Animated tamporal navigation`: the map
+  will be animated depending on the layer settings and temporal customization
```
[the remark this answers is not in the document]
Comment (ghtmtt) [reply docs/user_manual/temporal/temporal.rst:59]: yes right, not sure it is the right moment to describe it.

File: docs/user_manual/temporal/temporal.rst [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+main toolbar. Once clicked the temporal panel will appear:
+
+.. _temporal_controller:
+
+.. figure:: img/temporal_controller_panel.png
+   :align: center
+
+   Temporal Controller Panel
+
+The temporal panel has 3 main icons:
```
[the remark this answers is not in the document]
Comment (ghtmtt) [reply docs/user_manual/temporal/temporal.rst]: :)

File: docs/user_manual/temporal/temporal.rst (hunk 13 above)
Comment (ghtmtt) [reply docs/user_manual/temporal/temporal.rst]: This will be cleared when using vector layers

**Comment by ghtmtt** [comment]
many thanks @DelazJ for the review! I should have fixed all the main issues and added including the substitution names of the icons and the documentation guidelines. 

I can imagine you are a little bit confused. I started with raster data that need a little bit more tweaking than vector layers with a datetime field. Maybe @rduivenvoorde can have a look too and give his feedback? :)

We can always move sections (by sections I mean vectors, rasters, mesh, etc) step by step when we decide that some blocks are more clear for the final users. Starting a brand new section like this is quite difficult: when we agree with the structure (and icon names, etc) I will proceede smoother

**Comment by DelazJ** [comment]
superseded by #7021, #7022 and #7023
